pkg/subscriptions/subscriptions.go

```diff
@@ -385,3 +383,55 @@ func resolveSymbolicLink(path string) (string, error) {
 	}
 	return filepath.EvalSymlinks(path)
 }
+
+// mkdirAllIgnoreUmask creates a directory by ignoring the currently set umask.
+func mkdirAllIgnoreUmask(dir string, mode os.FileMode) error {
+	parent := dir
+	dirs := []string{}
+
+	// Find all parent directories which would have been created by MkdirAll
+	for {
+		if _, err := os.Stat(parent); os.IsNotExist(err) { //nolint: gocritic
+			dirs = append(dirs, parent)
```

Comment:
(This could be moved after this `if`. It’s not clear to me why this line, and not the other lines, belong inside / outside  the `if`)

Reply:
Yep, moved the code around to reduce the amount of else-ifs